services/agent/src/simcore_service_agent/services/volumes_manager.py

```diff
@@ -0,0 +1,179 @@
+import logging
+from asyncio import Lock, Task
+from dataclasses import dataclass, field
+from datetime import datetime, timedelta
+from typing import Final
+
+import arrow
+from aiodocker.docker import Docker
+from fastapi import FastAPI
+from models_library.projects_nodes_io import NodeID
+from pydantic import NonNegativeFloat
+from servicelib.background_task import start_periodic_task, stop_periodic_task
+from servicelib.logging_utils import log_context
+from tenacity import AsyncRetrying, before_sleep_log, stop_after_delay, wait_fixed
+
+from ..core.errors import NoServiceVolumesFoundError
+from ..core.settings import ApplicationSettings
+from .docker_utils import get_unused_dynamc_sidecar_volumes, remove_volume
+
+_logger = logging.getLogger(__name__)
+
+_WAIT_FOR_UNUSED_SERVICE_VOLUMES: Final[timedelta] = timedelta(minutes=1)
+
+
+@dataclass
+class VolumesManager:
```

Comment:
Why not use our friend SingletonInAppStateMixin ? This way setters and getters of app.state are all stardardized